components/search/src/filter.rs

```diff
@@ -108,11 +110,91 @@ fn maybe_extract_engine_config(
         .map(|_variant| SearchEngineDefinition::from_configuration_details(&identifier, base))
 }
 
+fn determine_default_engines(
+    engines: &[SearchEngineDefinition],
+    default_engines_record: Option<JSONDefaultEnginesRecord>,
+    user_environment: &SearchUserEnvironment,
+) -> (Option<String>, Option<String>) {
+    match default_engines_record {
+        None => (None, None),
+        Some(record) => {
+            let mut default_engine_id = None;
+            let mut default_engine_private_id = None;
+
+            let specific_default = record
+                .specific_defaults
+                .into_iter()
+                .rev()
+                .find(|r| matches_user_environment(&r.environment, user_environment));
+
+            if let Some(specific_default) = specific_default {
+                // Check the engine is present in the list of engines before
+                // we return it as default.
+                let engine_id = find_engine_with_match(engines, specific_default.default);
+                if engine_id.is_some() {
+                    default_engine_id = engine_id;
+                }
+                let private_engine_id =
+                    find_engine_with_match(engines, specific_default.default_private);
+                if private_engine_id.is_some() {
+                    default_engine_private_id = private_engine_id;
+                }
+            }
+
+            (
+                // If we haven't found a default engine in a specific default,
+                // then fall back to the global default engine - but only if that
+                // exists in the engine list.
+                //
+                // For the normal mode engine (`default_engine_id`), this would
+                // effectively be considered an error. However, we can't do anything
+                // sensible here, so we will return `None` to the application, and
+                // that can handle it.
+                default_engine_id.or_else(|| engine_id_if_found(engines, record.global_default)),
+                default_engine_private_id
+                    .or_else(|| engine_id_if_found(engines, record.global_default_private)),
+            )
+        }
+    }
+}
+
+fn engine_id_if_found(engines: &[SearchEngineDefinition], engine_id: String) -> Option<String> {
```

Comment:
What are your thoughts on renaming this function to `get_engine_id` or `find_engine_id`? 

I find the `engine_id_if_found` , specifically the `if_found` part repetitive because the `-> Option<String>` indicates it could return a String or None if not found.